src/lib/nftClaimer/deploy.ts

```diff
@@ -0,0 +1,100 @@
+import { getAddress } from '@ethersproject/address';
+import { splitSignature } from '@ethersproject/bytes';
+import { Interface } from '@ethersproject/abi';
+import { fetchSpace } from '../../helpers/snapshot';
+import { signer, validateAddresses, validateSpace } from './utils';
+import abi from './deployAbi.json';
+
+const DeployType = {
+  Deploy: [
+    { name: 'implementation', type: 'address' },
+    { name: 'initializer', type: 'bytes' },
+    { name: 'salt', type: 'uint256' }
+  ]
+};
+
+export default async function payload(
+  spaceOwner: string,
+  id: string,
+  maxSupply: number,
+  mintPrice: string,
+  proposerFee: number,
+  salt: string,
+  spaceTreasury: string
+) {
+  const space = await fetchSpace(id);
+  await validateSpace(spaceOwner, space);
+  if (proposerFee < 0 || proposerFee > 100) {
+    throw new Error('proposerFee should be between 0 and 100');
```

Comment:
Also could we move these too,move to before `fetchSpace`, like it is better to check the inputs first than sending the unwanted calls to hub